### PR TITLE
narcolepsy symptom fix

### PR DESCRIPTION
**Pull request by imsxz**
## About The Pull Request

changes narcolepsy from clamp to min
## Why It's Good For The Game

narcolepsy was giving people thousands of drowsiness. you were basically stuck passing out forever even if you cured it.
## Changelog
:cl:
fix: narcolepsy symptom doesnt give you infinite drowsiness anymore
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Why though? Just switch the statements in the clamp around.

File: code/datums/diseases/advance/symptoms/narcolepsy.dm [hunk 1]
```diff
@@ -63,7 +63,7 @@ Bonus
 		if(5)
 			if(prob(50))
 				to_chat(M, "<span class='warning'>[pick("So tired...","You feel very sleepy.","You have a hard time keeping your eyes open.","You try to stay awake.")]</span>")
-			M.drowsyness = CLAMP(M.drowsyness, M.drowsyness + 40, 70)
+			M.drowsyness = min(M.drowsyness + 40, 70)
```
Comment (nemvar) [inline code/datums/diseases/advance/symptoms/narcolepsy.dm]:
```suggestion
			M.drowsyness = CLAMP(M.drowsyness + 40, M.drowsyness, 70)
```

Comment (nemvar) [inline code/datums/diseases/advance/symptoms/narcolepsy.dm]:
```suggestion
			min(M.drowsyness + 40, max(70, M.drowsyness))
```

Comment (nemvar) [reply code/datums/diseases/advance/symptoms/narcolepsy.dm]: Or
```suggestion
		CLAMP(70, M.drowsyness + 40, M.drowsyness)
```
should work too if I'm not completely demented

**Comment by imsxz** [comment]
imo clamp doesnt have a place here. you dont wanna keep drowsiness between 2 numbers, you just wanna keep it less than 70.

**Comment by nemvar** [comment]
Mate, if drowsyness is above 70 from other sources, this will reduce it.

**Comment by ATH1909** [comment]
Uh, the clamp shouldn't be giving people thousands of drowsyness ticks, just up to 70 at a time.

File: code/datums/diseases/advance/symptoms/narcolepsy.dm (hunk 1 above)
Comment (ATH1909) [reply code/datums/diseases/advance/symptoms/narcolepsy.dm]: Wait, did I misunderstand how CLAMP works? I thought it just chose the non-highest, non-lowest value of the three values in the CLAMP, without caring about their ordering.

Comment (ATH1909) [reply code/datums/diseases/advance/symptoms/narcolepsy.dm]: oh god i did oh shit oh fuck hold on i need to make a big fix PR this is bad oh god

**Comment by ATH1909** [comment]
> Uh, the clamp shouldn't be giving people thousands of drowsyness ticks, just up to 70 at a time.

As it turns out, I had thought that clamps did something other than what they currently actually do, and this is in fact an issue. Fortunately, this issue isn't as bad as I thought that it was, since the narcolepsy symptom is the only place that I've used a CLAMP in a PR of mine so far.

I've made a PR that fixes this issue in a slightly better way than this PR does: https://github.com/tgstation/tgstation/pull/46684

**Comment by imsxz** [comment]
closed because #46684

**Pull request closed without merging**